zerver/tests/test_github.py

```diff
@@ -0,0 +1,103 @@
+import json
+
+import mock
+
+from zerver.lib.github import (
+    get_latest_github_release_download_link_for_platform,
+    get_latest_github_release_version_for_repo,
+    verify_release_download_link,
+)
+from zerver.lib.test_classes import ZulipTestCase
+from zerver.lib.test_helpers import MockPythonResponse
+
+
+class GitHubTestCase(ZulipTestCase):
+    def test_get_latest_github_release_version_for_repo(self) -> None:
+        response_text = json.dumps({"tag_name": "v5.4.3"})
+        with mock.patch("zerver.lib.github.requests.get", return_value=MockPythonResponse(response_text, 200)) as m:
+            self.assertEqual(get_latest_github_release_version_for_repo("zulip-desktop"), "5.4.3")
+            m.assert_called_once_with("https://api.github.com/repos/zulip/zulip-desktop/releases/latest")
+
+        response_text = json.dumps({"tag_name": "5.4.4"})
+        with mock.patch("zerver.lib.github.requests.get", return_value=MockPythonResponse(response_text, 200)) as m:
+            # Versions starting without v
+            self.assertEqual(get_latest_github_release_version_for_repo("zulip-mobile"), "5.4.4")
+            m.assert_called_once_with("https://api.github.com/repos/zulip/zulip-mobile/releases/latest")
+            m.reset_mock()
```

Comment:
No need to reset a mock that we’re already done with. The `with` block takes care of cleanup.